### ISSUE TITLE
[OSW2.access.04] Keyboard navigation items needing UX design

### ISSUE THREAD

**Issue by jessicatelles** [openstax/bit-backlog#24]
Trello: https://trello.com/c/6mdL73cK/350-osw2access04-keyboard-navigation-items-needing-ux-design

Description
--
In order to meet WCAG20 Level A, users must be able to navigate the site without a mouse. Tabbing between items is the standard form of navigation. Requirements are:

  * tab navigation between all elements on a page
  * current selected item must visually show it is selected (highlight, box around it) see screenshot for example from CNX
  * Enter key must trigger current item if it is a link, menu or clickable
  * user cannot be trapped in content using a plugin

**Related Documentation**

  * [H91: Using HTML form controls and links](https://www.w3.org/TR/WCAG20-TECHS/H91.html)
  * [SCR20: Using both keyboard and other device-specific functions](https://www.w3.org/TR/WCAG20-TECHS/SCR20.html)
  * [SCR35: Making actions keyboard accessible by using the onclick event of anchors and buttons](https://www.w3.org/TR/WCAG20-TECHS/SCR35.html)
  * [SCR2: Using redundant keyboard and mouse event handlers](https://www.w3.org/TR/WCAG20-TECHS/SCR2.html)
  * [G21: Ensuring that users are not trapped in content](https://www.w3.org/TR/WCAG20-TECHS/G21.html)

**Failures to Avoid**

  * [F54: Failure of Success Criterion 2.1.1 due to using only pointing-device-specific event handlers (including gesture) for a function](https://www.w3.org/TR/WCAG20-TECHS/F54.html)
  * [F55: Failure of Success Criteria 2.1.1, 2.4.7, and 3.2.1 due to using script to remove focus when focus is received](https://www.w3.org/TR/WCAG20-TECHS/F55.html)
  * [F42: Failure of Success Criteria 1.3.1, 2.1.1, 2.1.3, or 4.1.2 when emulating links](https://www.w3.org/TR/WCAG20-TECHS/F42.html)
  * [F10: Failure of Success Criterion 2.1.2 and Conformance Requirement 5 due to combining multiple content formats in a way that traps users inside one format type](https://www.w3.org/TR/WCAG20-TECHS/F10.html)

**Analysis**

  * **OS Web** has some highlighting on button text and links in the footer. The text changes color to blue. I'm not sure if that is too subtle
  * **About Us** has boxes around links, but none of the staff images or their bios are available via keyboard. I think the image should be highlighted and the bio available if the user hits enter. How should the bio be closed? use enter again?
  * **Blog** The author name is the same color as the links, but is not a link. This could be confusing.
  * **Subjects page** has problems. 
    * Cover is not highlighted on selection. Should Enter take the user to the book page or display the popup?
  * **Book page** 
    * Selection of link at top of the page traps user. They cannot navigate to any links and cannot get back to the top
    * PDF buttons are not highlighted and user cannot access X to close. This is the same for all sidebar links that display options

Acceptance
--
User can navigate the site without using a mouse. See items in Analysis section.

Impact
--
* **Severity**: [ ] High [ ] Medium  [ ] Low

Epic
--
https://docs.google.com/document/d/1Tu_85RG2BUkIfyKrH6PrTnWswh3qPld3uKefU6R1n90/edit#heading=h.s3ajugtc2x26

![Screenshot_from_2017-04-26_07-06-09.png](https://images.zenhubusercontent.com/5b7582ca1006d675626d97f0/17d1c927-d2b5-4b15-baa1-86af63bb1d4d)

Design
--
- [ ] UX - Design @ fredlindner1 --- olp:18,24,30 ar:0,27
- [ ] UX - User feedback testing @ fredlindner1 --- ar:0,18
- [ ] PO - UX design approval

Code
--
- [ ] FE Code FE function @ royjohnson7 -- olp:3,5,7 --- ar:0,6
- [ ] QA Review implemented functionality.
- [ ] FE Code design/style @ royjohnson7 -- olp:5,7,9 --- ar:0,8
- [ ] UX Review implemented design/style
- [ ] PO Review implemented design/style

Test
--
- [ ] QA Test prep @ gregfitch --- ar:0,2
- [ ] QA Test exec @ gregfitch --- ar:0,16

**Comment by jessicatelles** [comment]
Allison Howell Oct 12 at 2:28 PM:
Take to the team

Allison Howell Nov 13, 2017 at 9:53 AM:
@ Bruce we have determined that we can't get to this work on this 10K. Where should we move the card?

Bruce Pike Apr 27, 2017 at 10:20 AM:
Accepted. Break into separate stories for items that can be done now and items that need UX design.
